src/Microsoft.ML.Sweeper/Algorithms/Grid.cs

```diff
@@ -150,12 +150,12 @@ public RandomGridSweeper(IHostEnvironment env, Arguments args, IValueGenerator[]
             }
         }
 
-        public override ParameterSet[] ProposeSweeps(int maxSweeps, IEnumerable<IRunResult> previousRuns)
+        public override ParameterSet[] ProposeSweeps(int maxSweeps, IEnumerable<IRunResult> previousRuns = null)
         {
             if (_nGridPoints == 0)
                 return base.ProposeSweeps(maxSweeps, previousRuns);
 
-            var result = new List<ParameterSet>();
+            var result = new HashSet<ParameterSet>();
```

Comment:
> HashSet<ParameterSet>() [](start = 28, length = 24)

Looking at the other sweeping algos that we have. They might need the same fix. Will update the comment.

Reply:
The sweeper in KdoSweeper eventually uses SweeperBase which this PR already fixes. Yes, there are some other fixes to make, but probably better as another issue. I could look at them in a few days under a different issue

Reply:
Addressing them on a different PR, and logging something as a reminder sounds good to me. 

---
In reply to: [196239510](https://github.com/dotnet/machinelearning/pull/368#discussion_r196239510) [](ancestors = 196239510)